lib/common/socket.c

```diff
@@ -946,7 +946,8 @@ static void create_ossl(h2o_socket_t *sock)
 }
 
 static SSL_SESSION *on_async_resumption_get(SSL *ssl,
-#if OPENSSL_VERSION_NUMBER >= 0x1010000fL && !defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x1010000fL && !defined(LIBRESSL_VERSION_NUMBER)) || \
+    (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER > 0x2070000f)
```

Comment:
Thank you for the PR. Could this be changed to something like below? It is my understanding that conditional operators are supported by the C preprocessor.

```
#if defined(LIBRESSL_VERSION_NUMBER) ? LIBRESSL_VERSION_NUMBER > 0x2070000f : OPENSSL_VERSION_NUMBER >= 0x1010000fL
```